magento/inventory
# MSI-1614: Configurable product on custom stock doesn't disappear from category page on frontend when totally sold out.

### Description

Added cache invalidation for parent products together with child products.

### Fixed Issues (if relevant)

1. magento-engcom/msi#1614: Configurable product on custom stock doesn't disappear from category page on frontend when totally sold out.

### Manual testing scenarios

1. Create custom Stock with custom Source and assign it to Main website
2. Create configurable product with 2 simple variables - set the weight = 1 (red and blue) with qty = 10 on custom Source.
3. Go to Main website and place order with all qty of configurable product
    (red qty = 10 and blue qty = 10)
4. Go to admin and create shipment for your order
5. Navigate to storefront to category view page.
6. Configurable product should be absent on category view page.

### Contribution checklist
